### DISCUSSION TITLE
Can I customize the root partition disk size? Or mount a data volume at /nix?

### DISCUSSION THREAD

**Discussion by ghost** [lima-vm/lima#419] [Q&A] [answered]
👋 I'm exploring using lima and nix for a reproducible development environment.
I am running into the root partition filling up quickly (2GB).

Can I customize the root partition disk size? Or mount a data volume at /nix?

I am using alpine as the base image and then installing [nix](https://nixos.org/guides/install-nix.html).

Nix uses `/nix` to store everything related to itself and packaging. AFAIK, you cannot configure that.
That path quickly fills up:
```
df -h /nix
Filesystem                Size      Used Available Use% Mounted on
tmpfs                     1.9G      1.9G         0 100% /
```

Yet I see my configured 100GB of space in /tmp/data:
```
df -h /tmp/
Filesystem                Size      Used Available Use% Mounted on
/dev/disk/by-label/data-volume
                         97.9G    421.7M     92.5G   0% /mnt/data
```

Thanks!

**Comment by jandubois** [comment, marked as answer]
This is specific to Alpine (because we use an ISO and not a regular image):

The "diffdisk" in this case is indeed a separate data volume, and the following directories have been moved to the persisted disk: `/etc`, `/home`, `/tmp`,  `/usr/local`, and `/var/lib`.

You will notice that changes to any other part of the filesystem will be reverted by an instance restart because the OS runs from a `tmpfs`.

However, it should be easy enough to move your `/nix` directory to the persistent volume too:

```console
# mv /nix /mnt/data
# mkdir /nix
# mount --bind /mnt/data/nix /nix
```

You will also want to create a system provisioning script in your `lima.yaml` to repeat the `mount` command on each boot.
